File: 2024-2/Proyecto DAO SoftPub Java/SoftPubPersonaTest/src/softpubdbpersonatest/EspecialidadBOTest.java
package softpubdbpersonatest;

import java.util.ArrayList;
import pe.edu.pucp.softpub.persona.bo.EspecialidadBO;
import pe.edu.pucp.softpub.persona.model.Especialidad;
import pe.edu.pucp.softpub.persona.model.Nivel;

/**
 *
 * Autor: andres
 */
public class EspecialidadBOTest {

    private static EspecialidadBO especialidadBO;

    static void testEspecialidadBO() {
        System.out.println("\ntestEspecialidadBO");
        especialidadBO = new EspecialidadBO();
        testEspecialidadBOInsertar();
        testEspecialidadBOListarTodos();
        testEspecialidadBOModificar();
        testEspecialidadBOListarTodos();
        testEspecialidadBOObtenerPorId();
        testEspecialidadBOEliminar();        
    }

    private static void testEspecialidadBOEliminar() {
        System.out.println("\ntestEspecialidadBOEliminar");
        int resultado = especialidadBO.eliminar(1);
        resultado = especialidadBO.eliminar(2);
    }

    private static void testEspecialidadBOObtenerPorId() {
        System.out.println("\ntestEspecialidadBOObtenerPorId");
        Especialidad especialidad = especialidadBO.obtenerPorId(1);
        System.out.println("objetoPorId: 1 - " + especialidad);
        System.out.print(especialidad.getIdEspecialidad().toString());
        System.out.print(", ");
        System.out.print(especialidad.getEspecialidad());
        System.out.print(", ");
        System.out.print(especialidad.getFacultad());
        System.out.print(", ");
        System.out.println(especialidad.getNivel().toString());

        especialidad = especialidadBO.obtenerPorId(2);
        System.out.println("objetoPorId: 2 - " + especialidad);
        System.out.print(especialidad.getIdEspecialidad().toString());
        System.out.print(", ");
        System.out.print(especialidad.getEspecialidad());
        System.out.print(", ");
        System.out.print(especialidad.getFacultad());
        System.out.print(", ");
        System.out.println(especialidad.getNivel().toString());

        especialidad = especialidadBO.obtenerPorId(3);
        System.out.println("objetoPorId: 3 - " + especialidad);        
    }

    private static void testEspecialidadBOModificar() {
        System.out.println("\ntestEspecialidadBOModificar");
        int resultado = especialidadBO.modificar(1, "Ingeniería Informática", "FCI", Nivel.PREGRADO);
        resultado = especialidadBO.modificar(2, "Maestría en Informática", "EP", Nivel.MAESTRIA);
    }

    private static void testEspecialidadBOListarTodos() {
        System.out.println("\ntestEspecialidadBOListarTodos");
        ArrayList<Especialidad> lista = especialidadBO.listarTodos();
        for (Especialidad especialidad : lista) {
            System.out.print(especialidad.getIdEspecialidad().toString());
            System.out.print(", ");
            System.out.print(especialidad.getEspecialidad());
            System.out.print(", ");
            System.out.print(especialidad.getFacultad());
            System.out.print(", ");
            System.out.println(especialidad.getNivel().toString());
        }
    }

    private static void testEspecialidadBOInsertar() {
        System.out.println("\ntestEspecialidadBOInsertar");
        int resultado = especialidadBO.insertar(1, "Ingeniería Informática", "Ciencias e Ingeniería", Nivel.PREGRADO);
        resultado = especialidadBO.insertar(2, "Maestría en Informática", "Escuela de Posgrado", Nivel.MAESTRIA);
    }
}

File: 2024-2/0684/unidad07.f - Programación distribuida - reportes/ejemplo completo/SoftPub_Front-end/SoftPubPersonaTest/EspecialidadBOTest.cs
using System;
using System.ComponentModel;
using SoftPubBaseBO.ServicioWeb;
using SoftPubPersonaBO;

namespace SoftPubPersonaTest
{
    internal class EspecialidadBOTest
    {
        private static EspecialidadBO especialidadBO;

        internal static void testEspecialidadBO()
        {
            System.Console.WriteLine("testEspecialidadBO");
            especialidadBO = new EspecialidadBO();
            testEspecialidadBOInsertar();
            testEspecialidadBOListarTodos();
            testEspecialidadBOModificar();
            testEspecialidadBOListarTodos();
            testEspecialidadBOObtenerPorId();
            testEspecialidadBOEliminar();
        }

        private static void testEspecialidadBOEliminar()
        {
            System.Console.WriteLine("testEspecialidadBOEliminar");
            int resultado = especialidadBO.eliminar(1);
            resultado = especialidadBO.eliminar(2);
        }

        private static void testEspecialidadBOObtenerPorId()
        {
            System.Console.WriteLine("testEspecialidadBOObtenerPorId");
            especialidad _especialidad = especialidadBO.obtenerPorId(1);
            System.Console.WriteLine("objetoPorId: 1 - " + _especialidad);
            System.Console.Write(_especialidad.idEspecialidad);
            System.Console.Write(", ");
            System.Console.Write(_especialidad.especialidad1);
            System.Console.Write(", ");
            System.Console.Write(_especialidad.facultad);
            System.Console.Write(", ");
            System.Console.WriteLine(_especialidad.nivel);

            _especialidad = especialidadBO.obtenerPorId(2);
            System.Console.WriteLine("objetoPorId: 2 - " + _especialidad);
            System.Console.Write(_especialidad.idEspecialidad);
            System.Console.Write(", ");
            System.Console.Write(_especialidad.especialidad1);
            System.Console.Write(", ");
            System.Console.Write(_especialidad.facultad);
            System.Console.Write(", ");
            System.Console.WriteLine(_especialidad.nivel);

            _especialidad = especialidadBO.obtenerPorId(3);
            System.Console.WriteLine("objetoPorId: 3 - " + _especialidad);
            System.Console.Write(_especialidad.idEspecialidad);
            System.Console.Write(", ");
            System.Console.Write(_especialidad.especialidad1);
            System.Console.Write(", ");
            System.Console.Write(_especialidad.facultad);
            System.Console.Write(", ");
            System.Console.WriteLine(_especialidad.nivel);
        }

        private static void testEspecialidadBOModificar()
        {
            System.Console.WriteLine("testEspecialidadBOModificar");
            int resultado = especialidadBO.modificar(1, "Ingeniería Informática", "FCI", nivel.PREGRADO.ToString());
            resultado = especialidadBO.modificar(2, "Maestría en Informática", "EP", nivel.MAESTRIA.ToString());
        }

        private static void testEspecialidadBOListarTodos()
        {
            System.Console.WriteLine("testEspecialidadBOListarTodos");
            BindingList<especialidad> lista = especialidadBO.listarTodos();
            foreach (especialidad _especialidad in lista)
            {
                System.Console.Write(_especialidad.idEspecialidad);
                System.Console.Write(", ");
                System.Console.Write(_especialidad.especialidad1);
                System.Console.Write(", ");
                System.Console.Write(_especialidad.facultad);
                System.Console.Write(", ");
                System.Console.WriteLine(_especialidad.nivel);
            }
        }

        private static void testEspecialidadBOInsertar()
        {
            System.Console.WriteLine("testEspecialidadBOInsertar");
            int resultado = especialidadBO.insertar(1, "Ingeniería Informática", "Ciencias e Ingeniería", nivel.PREGRADO.ToString());
            resultado = especialidadBO.insertar(2, "Maestría en Informática", "Escuela de Posgrado", nivel.MAESTRIA.ToString());
        }
    }
}